wled00/fcn_declare.h

```diff
@@ -395,6 +401,23 @@ void enumerateLedmaps();
 uint8_t get_random_wheel_index(uint8_t pos);
 float mapf(float x, float in_min, float in_max, float out_min, float out_max);
 
+// fast (true) random numbers using hardware RNG, all functions return values in the range lowerlimit to upperlimit-1
+// note: for true random numbers with high entropy, do not call faster than every 200ns (5MHz)
+// tests show it is still highly random reading it quickly in a loop (better than fastled PRNG)
+// for 8bit and 16bit random functions: no limit check is done for best speed
+// 32bit inputs are used for speed and code size, limits don't work if inverted or out of range
+// inlining does save code size except for random(a,b) and 32bit random with limits
+#define random hw_random // replace arduino random()
```

Comment:
You should've used this approach for both `random8()` and `random16()` instead of replacing hundreds of function calls.
It will also replace those functions in library templates if they exist.

Reply:
that is what I had initially, but some FX require random8() and random16() unchanged. is there a way to use this #define but still have some functions use the original random8()/random16()  from fastled?

Reply:
Depends on the includes ant their use.

Reply:
Perhaps adding a wrapper of some kind would be the best approach.

On a second thought, it looks like there needs to be a distinction between *real* randomness and *pseudo* randomness if you want to have "super-sync" functionality. IMO it is a task to get these functions to explicitly use *pseudo* random functions.

Reply:
> You should've used this approach for both `random8()` and `random16()` instead of replacing hundreds of function calls.

Looking at code readability and maintainability, it would be better to replace at least the function calls in fx.cpp, fx_fcn.cpp, color.cpp and fx_2dfcn.cpp directly in the code. It will be a search & replace in the editor. If missing a few calls - so what. This will not cause errors because the old code "used to work".

We have some PRs already that "secretly replace" functions with optimized counterparts. I think it makes the code less readable at least for poeple who cannot remember all the  "secretly replacements".